### PR TITLE
feat: Github actions upload to S3

### PR DESCRIPTION
**Pull request by borichellow**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

_Briefly describe the issue you have solved or implemented with this pull request. If the PR contains multiple issues, use a bullet list._

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [4759](https://10.10.124.134/job/AR-build-pipeline/4759/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0(9681419)(9681419)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0(9681419)(9681419)-staging-compat.apk)

File: .github/workflows/upload-aws-s3.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+jobs:
+    deploy:
+        runs-on: ubuntu-latest
+        steps:
+            -   uses: jakejarvis/s3-sync-action@master
+                with:
+                    args: --acl public-read --follow-symlinks --delete
+                env:
+                    AWS_S3_BUCKET: "z-lohika"
```
Comment (yamilmedina) [inline .github/workflows/upload-aws-s3.yml]: We can use here `AWS_S3_BUCKET` added it to the repo too as a secret

File: .github/workflows/upload-aws-s3.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+on: push
+
+jobs:
+    deploy:
+        runs-on: ubuntu-latest
+        steps:
+            -   uses: jakejarvis/s3-sync-action@master
+                with:
+                    args: --acl public-read --follow-symlinks --delete
```
Comment (yamilmedina) [inline .github/workflows/upload-aws-s3.yml]: I didn't check, but we should keep `private`

File: .github/workflows/upload-aws-s3.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+                env:
+                    AWS_S3_BUCKET: "z-lohika"
+                    DEST_DIR: "megazord/android/reloaded/${params.FLAVOR.toLowerCase()}/${params.BUILD_TYPE.toLowerCase()}/"
```
Comment (yamilmedina) [inline .github/workflows/upload-aws-s3.yml]: As I understand from [here](https://github.com/wireapp/wire-android/blob/d4ec2b55a79258f20c2009be5e5419abb2ad3249/AR-builder.groovy#L391-L415) 

We should upload the APK to the path you use correctly in the PR, **plus** this other one as well (depending on [this](https://github.com/wireapp/wire-android/blob/d4ec2b55a79258f20c2009be5e5419abb2ad3249/AR-builder.groovy#L407) 

```
if (params.SOURCE_BRANCH.startsWith("PR-") || params.SOURCE_BRANCH == "develop" || params.SOURCE_BRANCH == "release/candidate") 
```

To (note the by-branch prefix)

path: "megazord/android/reloaded/**by-branch**/${params.SOURCE_BRANCH}/${params.FLAVOR.toLowerCase()}/${params.BUILD_TYPE.toLowerCase()}/"

**Review by yamilmedina — CHANGES_REQUESTED** [review]
Good to see this becoming a reality 🥇 
We might want to consider, using this action instead (suggestion) https://github.com/marketplace/actions/s3-upload-action

As provides as well a way to post the URL back to the PR, like we are doing today with Jenknins, ie. https://github.com/wireapp/wire-android/pull/3009#issuecomment-2112762165

**Comment by AndroidBob** [comment]
Build [4790](https://10.10.124.134/job/AR-build-pipeline/4790/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0(9697060)(9697060)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0(9697060)(9697060)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [4797](https://10.10.124.134/job/AR-build-pipeline/4797/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0(9697946)(9697946)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0(9697946)(9697946)-staging-compat.apk)

File: .github/workflows/testing-test-build-app.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        cache: gradle
+    - name: Build Staging Compat
+      env:
+        GITHUB_USER: ${{ github.actor }}
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        ./gradlew assembleStagingCompat -p ./ --no-daemon
+    - name: Upload APK staging compat
+      if: success()
+      uses: ./.github/workflows/upload-aws-s3.yml
```
Comment (yamilmedina) [inline .github/workflows/testing-test-build-app.yml]: Can you try to use, just for testing purposes, I have the feeling this is a temporary problem, and might be solved after merge to the default branch (`develop`)

```suggestion
      uses: ./.github/workflows/upload-aws-s3.yml@feat/github-actions-upload-to-s3
```

**Comment by AndroidBob** [comment]
Build [4799](https://10.10.124.134/job/AR-build-pipeline/4799/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0(9698643)(9698643)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0(9698643)(9698643)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [4838](https://10.10.124.134/job/AR-build-pipeline/4838/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0(9731763)(9731763)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0(9731763)(9731763)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [4841](https://10.10.124.134/job/AR-build-pipeline/4841/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0(9732111)(9732111)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0(9732111)(9732111)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [4861](https://10.10.124.134/job/AR-build-pipeline/4861/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0(9733621)(9733621)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0(9733621)(9733621)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [4867](https://10.10.124.134/job/AR-build-pipeline/4867/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0(9734330)(9734330)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0(9734330)(9734330)-staging-compat.apk)

**Pull request closed without merging**